sql/parser/eval.go

```diff
@@ -547,10 +540,12 @@ type EvalContext struct {
 	NodeID        uint32
 	StmtTimestamp DTimestamp
 	TxnTimestamp  DTimestamp
+	ReCache       *RegexpCache
 	GetLocation   func() (*time.Location, error)
 }
 
 var defaultContext = EvalContext{
+	ReCache: NewRegexpCache(128), // TODO(nvanbenschoten) how should this be sized?
```

Comment:
The `RegexpCache` is now safe to use through a nil reference, where it will act like a cache with no capacity. I also hardcoded the cache size to 512 entries per Ben's suggestion.